### ISSUE TITLE
Question regarding remote swap trace

### ISSUE THREAD

**Issue by shinyehtsai** [NetSys/disaggregation#2]
Hello Peter,

I have a question regarding remote swap trace. You mentioned that you implement a remote swap memory device and you use this swap memory to evaluate network requirement of resource disaggregation.

If I want to investigate the request trace of accessing remote swap memories (including accessing frequency, timestamp, and offset - I guess the size is always 4K?), which log files should I work on? 

There are NIC, DISK, META, and MEM. Would you mind to illustrate a little bit more about different traces?

If I want to see remote swap request, should I work on DISK or MEM?

Thank you very much.
Shin-Yeh

**Comment by pxgao** [comment]
> Hello Peter,
> 
> I have a question regarding remote swap trace. You mentioned that you implement a remote swap memory device and you use this swap memory to evaluate network requirement of resource disaggregation.
> 
> If I want to investigate the request trace of accessing remote swap memories (including accessing frequency, timestamp, and offset - I guess the size is always 4K?), which log files should I work on?
Yes. Always 4K, you should use the MEM trace if you are only interested in the remote swap.

> There are NIC, DISK, META, and MEM. Would you mind to illustrate a little bit more about different traces?
NIC: We assume that traditional NIC traffic will still exist for legacy applications on disaggregated datacenter. NIC is the measuring this part.
DISK: Is the traffic going to disk. In disaggregated datacenter, it will go to remote disk
META: Only contains the timestamp when the experiment starts. (in us)
MEM is the memory trace to simulated remote swap.

> If I want to see remote swap request, should I work on DISK or MEM?
MEM

> Thank you very much.
> Shin-Yeh

**Comment by shinyehtsai** [comment]
Hello Peter,

Thank you for your response. I am working on mem trace now. When I was investigating the trace, I have two questions, and I was hoping you could help me to figure out.

I found out that the request size is always multiple pages and sometimes larger than 100 or even more. Is this a single swap memory request? If this is a single request, is this a blocking call? Or this is a batch request (request size is the summation of all the requests within a single batch)?

I found out that the default remote memory size is 22.09 GB. Why did you pick this number? And what is the local memory size?

Thank you for your help and time.

Shin-Yeh

**Comment by pxgao** [comment]
I found out that the request size is always multiple pages and sometimes larger than 100 or even more. Is this a single swap memory request? If this is a single request, is this a blocking call? Or this is a batch request (request size is the summation of all the requests within a single batch)?
OS will prefer to batch the requests. That is a result of paging system of OS.

We assume 25% local memory in most of the setting. 22.09 should be 75% of the available total memory on the experiment machine.
